form-test-tom/src/App.jsx

```diff
@@ -0,0 +1,105 @@
+import {useForm} from './useForm'
+import React from 'react'
+import params from './formParams.js'
+
+const animaux = ['bear','tiger','snake','donkey']
+
+const TorchboxForm = () => {
+  const { handleSubmit, handleChange, data, errors } = useForm(params)
```

Comment:
a custom state hook which takes a set of form options, which make this hook reusable